### DISCUSSION TITLE
Lerna v7 no longer does package dependency management

### DISCUSSION THREAD

**Discussion by subzero10** [honeybadger-io/honeybadger-js#1229] [General]
I'm opening a discussion because we must first do some research, discuss available options and then proceed to implementation.

We are on currently Lerna v6.
In order to upgrade to v7, we will have to use a new dependency manager for the monorepo:
- npm workspaces
- yarn workspaces
- pnpm workspaces (or bit.dev)
- nx

Personally, I have tried npm workspaces and quickly moved away from them, as they are not as flexible and can lead to a dead end. I even came up with a [demo project](https://github.com/subzero10/npm-workspaces-demo) recently to demonstrate the issue.

At this moment, I am experimenting with pnpm and it seems pretty powerful.

Finally, the latest versions of node.js come with a beta feature [corepack](https://nodejs.org/api/corepack.html) allowing better integration with different package managers.

If you have any thoughts or feedback to share on the topic, please share them here.

**Comment by BethanyBerkowitz** [comment]
Just adding the link to the [docs from Lerna](https://lerna.js.org/docs/legacy-package-management) for later since they're pretty helpful. 

I've heard great things about pnpm, though I haven't used it yet. This was very encouraging: https://dev.to/hotjar/monorepos-with-pnpm-part-1-a-performant-package-manager-5g41

The bug you demonstrated with the dependencies getting hoisted up is worrisome -- you could be developing with your package having access to a dependency that isn't actually in its `package.json`, and you probably wouldn't notice. Thanks for pointing this out!

**Comment by subzero10** [comment]
Some issues I came across setting up a new PC:
- https://github.com/lerna/lerna/issues/2447
- https://github.com/lerna/lerna/issues/274
